### PR TITLE
allow setting of user/password to authorize (other than using 'code').

### PR DESCRIPTION
**Pull request by yuhanz**
I just added an alternative way to authorize by directing giving user/password without getting into the redirect url login and get 'code'. The weibo php client has the feature, but this python client is missing it. So I'm adding it to serve for my purpose.

### CODE REVIEW

File: weibo.py [hunk 1]
```diff
@@ -17,7 +16,7 @@
 
 
 class Client(object):
-    def __init__(self, api_key, api_secret, redirect_uri, token=None):
+    def __init__(self, api_key, api_secret, redirect_uri, token=None, user=None, password=None):
```
Comment (lxyu) [inline weibo.py]: `username` is better.

File: weibo.py [hunk 2]
```diff
@@ -28,6 +27,8 @@ def __init__(self, api_key, api_secret, redirect_uri, token=None):
         self.client_id = api_key
         self.client_secret = api_secret
         self.redirect_uri = redirect_uri
+        self.user = user
```
Comment (lxyu) [inline weibo.py]: Is it possible to bind `auth` to session? for example:

``` python
if username and password:
    self.session.auth = username, password
```

Comment (yuhanz) [inline weibo.py]: changed accordingly

File: weibo.py (hunk 1 above)
Comment (yuhanz) [inline weibo.py]: changed accordingly

File: weibo.py [hunk 3]
```diff
@@ -30,6 +29,8 @@ def __init__(self, api_key, api_secret, redirect_uri, token=None):
         self.redirect_uri = redirect_uri
 
         self.session = requests.session()
+        if username and password:
+            self.session.auth = (username, password)
```
Comment (lxyu) [inline weibo.py]: the parentheses can be removed

File: weibo.py [hunk 4]
```diff
@@ -90,6 +91,7 @@ def get(self, uri, **kwargs):
         """Request resource by get method.
         """
         url = "{0}{1}.json".format(self.api_url, uri)
+        kwargs['source'] = self.client_id
```
Comment (lxyu) [inline weibo.py:94]: I assume the source only works if `session.auth` provided? add a `if` statement would be better.

File: weibo.py [hunk 5]
```diff
@@ -8,7 +8,6 @@
 For more info, refer to:
 http://lxyu.github.io/weibo/
 """
-
```
Comment (lxyu) [inline weibo.py:11]: let's keep the empty line here. ;)

File: weibo.py (hunk 3 above)
Comment (yuhanz) [inline weibo.py]: k

File: weibo.py (hunk 4 above)
Comment (yuhanz) [inline weibo.py:94]: I thinkg this doesn't hurt.

**Comment by lxyu** [comment]
released to pypi. you may use it now.

thanks very much for your contribution!

**Comment by yuhanz** [comment]
thanks man

**Pull request merged**